### ISSUE TITLE
generate the fields in _form.php file in the crud generetor based on the rules for the model and not the database table

### ISSUE THREAD

**Issue by Faryshta** [yiisoft/yii2-gii#57]
specially since sometimes I change the rules of the model before generating the crud. Some use cases

```php
[['file'], 'file'], // the form this will be a text input because on the column type is varchar
['active', 'boolean'], // will be a text input because the column type is int(1)
['delivery_date', 'date],  // will genereate text input
['amount', 'integer'], // will genereate text input, it can easily generate an input('number') instead
['email', 'email'], // will genereate text input, it can easily generate an input('email') instead
['category_id', 'exist', 'targetClass' => Category::className()], // another text input, a select would be much more useful
// on the other hand, created_at and updated_at will be generated even if they are not on the rules.
```

**Comment by samdark** [comment]
Could be a good idea.

**Comment by samdark** [comment]
Want to work on implementing it?

**Comment by Faryshta** [comment]
yes i can work on it. I would like to figure out a way to give priorities to rules.

For example. If there is an 'exist' foreign key then the best would be to give that key the maximum priority over any other rule and generate a drop down field. Since this will be for yii 2.0.5 we can use 

`$targetClass;;find()->select(['name'])->indexBy('id')->column()`

Or something similar, the issue would be when there is no `name`columns or when dealing with composite foreign keys.

another one that should get priority is `file` validator which would be pretty straight forward, simply use the `fileInput` method.

what other priorities you can think?

**Comment by Faryshta** [comment]
@samdark hi, i am working on this right now, I have the same questions as before, any decision there?

**Comment by Faryshta** [comment]
I am thinking about using http://www.yiiframework.com/doc-2.0/yii-base-model.html#getValidators%28%29-detail to get all the validators. Then assign a weight to each validator according to its class.

Each validated attribute will be assigned the validation with the biggest weight that applies to it.

For example if `category_id` as validations 'require', 'integer' and 'exists' the 'exists' validation will be used and will generate a dropdown list.

WIth this we can create CRUDS for models that are not active records.

I will require to modify the `generateActiveField` method and all the unit testings it implies. What do you think?

**Comment by samdark** [comment]
That's an interesting idea. Do you have a sketch for validator priorities?

**Comment by Faryshta** [comment]
exists => 4
file => 4
captcha => 4
image => 4
date => 3 // hard  to decide if put a calendar here for the jui requirement
in => 3 // maybe this should generate a drop down, but the allowed values can be dynamic
password => 3 // this will be from the input name
boolean => 2 // checkbox
email => 2 // ->input('email')
integer => 2 // ->input('number')
string = 1 //->textInput() default

all others (like unique and required) are not worth to make a difference on the input generated.

**Comment by samdark** [comment]
`default` should be 5

**Comment by Faryshta** [comment]
what input you would generate based on a `default` validation?

**Comment by samdark** [comment]
It makes sense to have `default null` rule for nullable fields.

**Comment by Faryshta** [comment]
yes but on the `_form.php` file what would you generate based on that rule?

**Comment by samdark** [comment]
Nothing.

**Comment by Faryshta** [comment]
well i don't understand what you mean with default yet and why it should be more important than the rest.

Do you want me to make a patch based on what I said and then you can add your `default` logic after?

**Comment by samdark** [comment]
No need to add anything about `default`. I've missed context when I was talking about it.

Yes, a patch would be great.
